packages/ingest-mongodb-public/src/sources/snooty/SnootyDataSource.ts

```diff
@@ -126,15 +126,18 @@ export type SnootyMetadata = {
   title?: string;
 };
 
-export type SnootyProjectConfig = ProjectBase & {
-  type: "snooty";
-  branches?: Branch[];
-};
+export type SnootyProjectConfig<ProductName extends string = string> =
+  ProjectBase<ProductName> & {
+    type: "snooty";
+    branches?: Branch[];
+  };
```

Comment:
rather than using the generic in `ProjectBase.ts`, you could just do:

```suggestion
  ProjectBase & {
    type: "snooty";
    branches?: Branch[];
    productName: MongoDbProduct;
  };
```

this would be a bit cleaner b/c you dont need to pass around the generic abstraction throughout the project when its not being used elsewhere.